codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res

```diff
@@ -257,25 +257,63 @@ let popBatchItem = (
     switch maybeArbItem {
     //If there is item on the arbitray events queue, and it is earlier than
     //than the earlist event, take the item off from there
-    | Some({val: itemWithPopFn, isInReorgThreshold})
-      if Item(itemWithPopFn)->getQueueItemComparitor(~chain=itemWithPopFn.item.chain) <
-        earliestEvent->getQueueItemComparitor(~chain) => {
+    | Some({val: arbVal, isInReorgThreshold})
+      if if arbVal.item.chain === chain {
```

Comment:
`if if` 😅

Reply:
This is tripping me up 😅 can we not extract the inner bool expression to a variable separate it into two expressions?